src/near/index.ts

```diff
@@ -0,0 +1,385 @@
+      chainIndex +
+      addressIndex +
+      contractDummyPadding +
+      '01' +
+      acc.padEnd(66, '0')
+    );
+  }
+
+  async getTotalBalanceLinked() {
+    const accounts = await getAccounts(this.nearPublicKey, this.coin.network);
+    const balances = await Promise.all(
+      accounts.map(async (account: any) => {
+        const balance = await getBalance(account.account_id, this.coin.network);
+        return balance;
+      })
+    );
+    const totalBalance = balances.reduce(
+      (acc: number, curr: number) => acc + curr,
+      0
+    );
+    return {
+      balance: totalBalance
+    };
+  }
+
+  async getTotalBalance() {
+    const balance = await getBalance(this.nearPublicKey, this.coin.network);
+    return { balance };
+  }
+
+  async getTotalBalanceCustom(account?: string) {
+    const balance = await getBalance(
+      account ? account : this.address,
+      this.coin.network
+    );
+    return { balance };
+  }
+
+  async setupNewWallet() {
+    // do nothing
+  }
+
+  async generateMetaData(gasFees: number, addAccount?: boolean) {
+    try {
+      logger.info('Generating metadata for near', {
+        address: this.address
+      });
+      const purposeIndex = '8000002c';
+      const coinIndex = this.coin.coinIndex;
+      const accountIndex = '80000000';
+
+      const inputCount = 1;
+      const chainIndex = '80000000';
+      const addressIndex = '80000001'; //could be changed to use this.node
+      const inputString = chainIndex + addressIndex;
+
+      const outputCount = 1;
+      const outputString = '0000000000000000';
+
+      const changeCount = 1;
+      const changeString = '0000000000000000';
+
+      const gas = intToUintByte(0, 32);
+
+      const decimal = intToUintByte(this.coin.decimal, 8);
+      const contractDummyPadding = intToUintByte(
+        Math.round(gasFees / 10000),
+        16 * 4
+      ); //changing decimal to fit the size only until protobuff
+      return (
+        purposeIndex +
+        coinIndex +
+        accountIndex +
+        intToUintByte(inputCount, 8) +
+        inputString +
+        intToUintByte(outputCount, 8) +
+        outputString +
+        intToUintByte(changeCount, 8) +
+        changeString +
+        gas +
+        decimal +
+        contractDummyPadding +
+        (addAccount ? intToUintByte(1, 8) : intToUintByte(0, 8))
+      );
+    } catch (e) {
+      logger.error('Error generating metadata', e);
+      throw e;
+    }
+  }
+
+  async generateUnsignedTransaction(
+    recieverAddress: string,
+    amount: BigNumber,
+    senderAddressArg?: string
+  ): Promise<{
+    txn: string;
+    inputs: Array<{
+      value: string;
+      address: string;
+      isMine: boolean;
+    }>;
+    outputs: Array<{ value: string; address: string; isMine: boolean }>;
+  }> {
+    try {
+      const senderAddress = senderAddressArg || this.address;
+      //@ts-ignore required to convert BigNumber to BN.js instance
+      const bnAmount = Web3.utils.toBN(amount);
```

Comment:
Why can't we do 

```
      const bnAmount = Web3.utils.toBN(amount.toString());
```

Reply:
I tried this it doesn't works

Reply:
for proper conversion, we need to pass the BN.js object. tostring for large numbers messes up the conversion for example if the amount is 1e+24 the string would be '1e+24' and the converter will treat is as a hex string for some reason